File: ps-frontend/src/model/call.ts
import {Status} from './status';
import {Location} from './location';
import {Order} from './order';
import {Note} from './note';
import {Group} from './group';
import {Person} from './person';
import {User} from './user';
import {Document} from './document';
import { MainCategory } from './main-category';

export class Call {
  public id?: number;
  public description?: string;
  public dateCreated?: Date;
  public dateEnded?: Date;

  public mainCategory?: MainCategory;

  public documents?: Document[];
  
  public informCaller?: boolean;
  public priority?: boolean;
  public confidential?: boolean;
  public channelInfo?: string;
  public channel?: string;
  public callcenter?: string;
  public team?: string;
  public casenumber?: string;
  public status?: Status;
  
  public location?: Location;
  public person?: Person;
  public orders?: Order[];
  public notes?: Note[];
  public user?: User;
  public createdBy?: User;
  public group?: Group;
  public city?: string;
  public postal?: string;
  public street?: string;
  public number?: string;
  public notification?: string;
}

File: ps-backend/src/main/java/org/commonground/ps/backendapi/model/Call.java
package org.commonground.ps.backendapi.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Call {
  private Long id;
  @NotNull(message = "Waarde is verplicht")
  @Size(min = 1, max = 500, message = "Waarde is minimaal 1 en maximaal 500 tekens")
  private String description;
  //@JsonFormat(pattern = "yyyy-MM-dd hh:mm:ss Z", timezone = "Europe/Amsterdam")
  private Date dateCreated;
  //@JsonFormat(pattern = "yyyy-MM-dd hh:mm:ss Z", timezone = "Europe/Amsterdam")
  private Date dateEnded;
  private String casenumber;

  @Valid
  @NotNull(message = "Waarde is verplicht")
  private MainCategory mainCategory;

  @Valid
  @NotNull(message = "Waarde is verplicht")
  private Location location;
  @Valid
  private Person person;
  private Domain domain;

  private Status status;

  private Group group;
  private User user;

  private List<Order> orders = new ArrayList<>();
  private List<Note> notes = new ArrayList<>();
}
